Library/Homebrew/formula.rb

```diff
@@ -1843,7 +1843,12 @@ def eligible_kegs_for_cleanup
       eligible_kegs = if head? && (head_prefix = latest_head_prefix)
         installed_kegs - [Keg.new(head_prefix)]
       else
-        installed_kegs.select { |k| pkg_version > k.version }
+        installed_kegs.select do |keg|
+          tab = Tab.for_keg(keg)
+          next true if version_scheme > tab.version_scheme
```

Comment:
Think it might be more readable to do a `if`, `elseif`, `else` on `>`, `==`, `<`